### PR TITLE
[feat]: [CDS-72877]: Addition of yaml field for service overrides v2

### PR DESCRIPTION
**Pull request by lovish1999**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

File: internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go [hunk 1]
```diff
@@ -17,39 +17,11 @@ func DataSourceServiceOverrides() *schema.Resource {
 		ReadContext: dataSourceServiceOverridesRead,
 
 		Schema: map[string]*schema.Schema{
-			"service_id": {
-				Description: "The service ID to which the overrides applies.",
+			"newly_created": {
+				Description: "Boolean field to check if the override entity is newly created.",
```
Comment (tchaurasiya) [inline internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go:21]: we dont need this.

File: internal/service/platform/service_overrides_v2/data_source_service_overrides_v2_test.go [hunk 2]
```diff
@@ -91,45 +93,29 @@ func TestAccDataSourceServiceOverrides(t *testing.T) {
...
-		                    "type": "String",
-		                    "value": "val1"
-		                  }
-		                ]
-		              }
-		              EOT
-				}
+                    yaml = <<-EOT
+                    {
+                        "variables": [
```
Comment (tchaurasiya) [inline internal/service/platform/service_overrides_v2/data_source_service_overrides_v2_test.go:103]: can we add manifest/config files too.

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go [hunk 3]
```diff
@@ -32,32 +30,42 @@ func ResourceServiceOverrides() *schema.Resource {
...
 				Required:    true,
 			},
-			"spec": {
-				Description: "The overrides specification for the service.",
+			"yaml": {
+				Description: "The yaml of the overrides spec object.",
 				Type:        schema.TypeString,
 				Required:    true,
 			},
+			"newly_created": {
```
Comment (tchaurasiya) [inline internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go:59]: this field is not needed

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go [hunk 4]
```diff
@@ -32,32 +30,42 @@ func ResourceServiceOverrides() *schema.Resource {
...
+				Description: "Boolean field to check if the override entity is newly created.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"identifier": {
```
Comment (tchaurasiya) [inline internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go:64]: we dont support identifier now?

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go [hunk 5]
```diff
@@ -129,46 +145,33 @@ func resourceServiceOverridesV2Delete(ctx context.Context, d *schema.ResourceDat
...
 	d.Set("project_id", so.ProjectIdentifier)
 	d.Set("env_id", so.EnvironmentRef)
 	d.Set("service_id", so.ServiceRef)
 	d.Set("infra_id", so.InfraIdentifier)
 	d.Set("cluster_id", so.ClusterIdentifier)
 	d.Set("type", so.Type_)
-	d.Set("spec", so.Spec)
-
+	d.Set("yaml", so.YamlInternal)
+	d.Set("newly_created", so.NewlyCreated)
```
Comment (tchaurasiya) [inline internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go:170]: can remove newly_created

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2_test.go [hunk 6]
```diff
@@ -190,14 +187,13 @@ func testAccServiceOverridesProjectScope(id string, name string) string {
...
-            type = "ENV_SERVICE_OVERRIDE"
-            spec = <<-EOT
+        resource "harness_platform_service_overrides_v2" "test" {
+          org_id     = harness_platform_organization.test.id
+          project_id = harness_platform_project.test.id
+          env_id     = harness_platform_environment.test.id
+          service_id = harness_platform_service.test.id
+          type       = "ENV_SERVICE_OVERRIDE"
+          yaml = <<-EOT
               {
```
Comment (tchaurasiya) [inline internal/service/platform/service_overrides_v2/resource_service_overrides_v2_test.go:197]: lets add manifest/config example

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go (hunk 4 above)
Comment (lovish1999) [reply internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go:64]: This is not for passing in the terraform plan. It is computed field. So users can use it to get the identifier value after the resource is created.

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go (hunk 5 above)
Comment (lovish1999) [reply internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go:170]: ok

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go (hunk 3 above)
Comment (lovish1999) [reply internal/service/platform/service_overrides_v2/resource_service_overrides_v2.go:59]: ok

File: internal/service/platform/service_overrides_v2/data_source_service_overrides_v2_test.go (hunk 2 above)
Comment (lovish1999) [reply internal/service/platform/service_overrides_v2/data_source_service_overrides_v2_test.go:103]: ok

File: internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go (hunk 1 above)
Comment (lovish1999) [reply internal/service/platform/service_overrides_v2/data_source_service_overrides_v2.go:21]: ok

File: internal/service/platform/service_overrides_v2/resource_service_overrides_v2_test.go (hunk 6 above)
Comment (lovish1999) [reply internal/service/platform/service_overrides_v2/resource_service_overrides_v2_test.go:197]: ok

**Review by tchaurasiya — APPROVED** [review]

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**